src/Metadata/AbstractMetadataFactory.php

```diff
@@ -0,0 +1,25 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive-hal for the canonical source repository
+ * @copyright Copyright (c) 2017 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive-hal/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Zend\Expressive\Hal\Metadata;
+
+/**
+ * TODO Would you prefer an interface?
```

Comment:
Yes. This is essentially an interface already, as it's only defining abstract methods. Let's make it an interface instead.

Reply:
Done.